sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/PartitionScopeThresholds.java

```diff
@@ -46,9 +46,15 @@ public PartitionScopeThresholds(String pkRangeId, CosmosBulkExecutionOptionsImpl
         this.maxMicroBatchSize = Math.min(
             options.getMaxMicroBatchSize(),
             BatchRequestResponseConstants.MAX_OPERATIONS_IN_DIRECT_MODE_BATCH_REQUEST);
+        this.minTargetMicroBatchSize = Math.max(
+            options.getMinTargetMicroBatchSize(),
+            Configs.getMinTargetBulkMicroBatchSize()
```

Comment:
Do we need another guard rail in case `Configs.getMinTargetBulkMicroBatchSize()` returns something functionally weird like -ve values or greater than 100?

Reply:
Yes, we should enforce maxBatchSize guard rail. Good catch - amde those changes.